documentation/api/io.strimzi.api.kafka.model.connect.build.Build.adoc

```diff
@@ -0,0 +1,191 @@
+The container registry is configured in the `.spec.build.output` section of the `KafkaConnect` custom resource.
+The `output` configuration supports two types: `docker` and `imagestream`.
+The `output` section is required.
+
+.Using Docker registry
+
+To use a Docker registry, you have to specify the `type` as `docker`, and the `image` field with the full name of the new container image.
+The full name has to include:
+
+* The address of the registry.
+* Port number if it is listening on non-standard port
+* The tag of the new container image
+
+Examples of valid image names are:
+
+* `docker.io/my-org/my-image/my-tag`
+* `quay.io/my-org/my-image/my-tag`
+* `image-registry.image-registry.svc:5000/myproject/kafka-connect-build:latest`
+
+Different Kafka Connect deployments should use a different image (for example at least with different tags).
+
+If the registry requires authentication, the field `pushSecret` can be used to set a name of the Secret with the registry credentials.
+The Secret should use the `kubernetes.io/dockerconfigjson` type and contain the `.dockerconfigjson` file with the Docker credentials.
+For more details see link:https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/#registry-secret-existing-credentials[Create a Secret based on existing Docker credentials^]
+
+[source,yaml,subs=attributes+,options="nowrap"]
+.Example `output` configuration
+----
+apiVersion: {KafkaConnectApiVersion}
+kind: KafkaConnect
+metadata:
+  name: my-connect-cluster
+spec:
+  #...
+  build:
+    output:
+      type: docker # <1>
+      image: my-registry.io/my-org/my-connect-cluster:latest # <2>
+      pushSecret: my-registry-credentials # <3>
+  #...
+----
+<1> The type of the output which should be used by Strimzi. (Required)
+<2> The full name of the image which will be used including the repository and tag. (Required)
+<3> Name of the secret with the container registry credentials. (Optional)
+
+.Using OpenShift ImageStream
+
+Alternatively, you can also use OpenShift ImageStream to store the new container image.
+The ImageStream has to be created manually before deploying the Kafka Connect.
+To use OpenShift ImageStream, you have to set the `type` to `imagestream`, and the `image` field should contain the name of the ImageStream and the tag which should be used.
+For example `my-connect-image-stream:latest`.
+
+[source,yaml,subs=attributes+,options="nowrap"]
+.Example `output` configuration
+----
+apiVersion: {KafkaConnectApiVersion}
+kind: KafkaConnect
+metadata:
+  name: my-connect-cluster
+spec:
+  #...
+  build:
+    output:
+      type: imagestream # <1>
+      imageStream: image-stream-name # <2>
+  #...
+----
+<1> The type of the output which should be used by Strimzi. (Required)
+<2> The name of the ImageStream and tag. (Required)
+
+=== `plugins`
+
```

Comment:
Connector plugins are a set of JAR files that define the implementation required to connect to certain types of external system.